kad-dht/README.md

```diff
@@ -0,0 +1,453 @@
+# libp2p Kademlia DHT specification
+
+| Lifecycle Stage | Maturity       | Status | Latest Revision |
+|-----------------|----------------|--------|-----------------|
+| 3A              | Recommendation | Active | r0, 2021-05-07  |
+
+Authors: [@raulk], [@jhiesey], [@mxinden]
+
+Interest Group:
+
+[@raulk]: https://github.com/raulk
+[@jhiesey]: https://github.com/jhiesey
+[@mxinden]: https://github.com/mxinden
+
+See the [lifecycle document][lifecycle-spec] for context about maturity level
+and spec status.
+
+[lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
+
+---
+
+## Overview
+
+The Kademlia Distributed Hash Table (DHT) subsystem in libp2p is a DHT
+implementation largely based on the Kademlia [0] whitepaper, augmented with
+notions from S/Kademlia [1], Coral [2] and the [BitTorrent DHT][bittorrent].
+
+This specification assumes the reader has prior knowledge of those systems. So
+rather than explaining DHT mechanics from scratch, we focus on differential
+areas:
+
+1. Specialisations and peculiarities of the libp2p implementation.
+2. Actual wire messages.
+3. Other algorithmic or non-standard behaviours worth pointing out.
+
+For everything else that isn't explicitly stated herein, it is safe to assume
+behaviour similar to Kademlia-based libraries.
+
+Code snippets use a Go-like syntax.
+
+## Definitions
+
+### Replication parameter (`k`)
+
+The amount of replication is governed by the replication parameter `k`. The
+default value for `k` is 20.
```

Comment:
When we use default throughout this document what do we mean? Is it the IPFS DHT (i.e. only applies to network parameters), the defaults in go-libp2p-kad-dht, or something else? If it's go-libp2p-kad-dht then  how problematic is it if some of the algorithms/query approaches are different then the ones mentioned here?

Reply:
Maybe recommended is better.

What I was trying to distinguish was between network-wide recommendations (e.g. k=20) and implementation-specific recommendations (e.g. alpha = 3).

My thinking was that network-wide parameters are much easier to give recommendations for. Client stuff is more a "here's one way to do it". Maybe this distinction is too in the weeds though. Your call